cloud-core/src/main/java/cloud/commandframework/Command.java

```diff
@@ -318,6 +328,28 @@ public Command(
         return new ArrayList<>(this.arguments);
     }
 
+    /**
+     * Return a copy of the command argument array, ignoring flag arguments.
+     *
+     * @return Copy of the command argument array, ignoring flag arguments.  This List is mutable.
+     */
+    public @NonNull List<CommandArgument<@NonNull C, @NonNull ?>> getActualArguments() {
```

Comment:
I see two options here
- 1
   - make getArguments return a legacy view of the arguments (flags at end only)
   - add rawArguments to get the raw arguments with potential repeated flags
- 2
   - check all internal usages (ie help) and make it properly handle the new actualArguments and flagArgument instead of relying on getArguments
   - expect users understand they may need to change other code when enabling flags anywhere
 
I'm not sure if other places like syntax messages and help menus need updating for "correct" behavior when the tree has flagnodes throughout (ie none of the aforementioned options address accessing the command tree, only the arguments through the command instance), so that probably also needs to be addressed.

Reply:
To maintain backwards-compat and not break API, i've made getArguments as well as the arguments inside stay the same, with flags at the end

The new method (getActualArguments) is the list of arguments without the flags, which is just a util to ignore flags. The getArguments method isn't considered a "legacy view" but rather still the "normal" view where flags will be at the end.

The flags being available between all args is only a detail relevant to the command tree registration, but the help generation/syntax will never display the command as having the flags everywhere because it isn't the case.

I believe this solution makes it so most api users can stay ignorant of the implementation detail that parsing will support flags everywhere, help menus etc would look absurdly verbose if they were displayed as such.

Reply:
This sounds reasonable to me